app/controllers/api/v1/dependencies_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Api::V1::DependenciesController < Api::BaseController
+  skip_before_action :verify_authenticity_token
+  GEM_REQUEST_LIMIT = 250
+
+  def index
+    return render text: '' if gem_names.empty?
+    return render text: "Too many gems!", status: 422 if gem_names.size > GEM_REQUEST_LIMIT
+
+    deps = gem_names.each_with_object([]) do |gem_name, deps_array|
+      gem = Rubygem.includes(:versions).find_by_name!(gem_name)
+      gem_deps_hashes = gem.versions.sort_by(&:number).reverse_each.map do |version|
+        version_deps = version.dependencies.find_all { |d| d.scope == "runtime" }
```

Comment:
you are right @segiddins , this will generate a N+1 query... indeed is a problem.